syntax = "proto3";

package borrowservice;

service BorrowService {
  rpc is_borrowed(IsBorrowedRequest) returns (IsBorrowedResponse);
  rpc is_borrowed_by(IsBorrowedByRequest) returns (IsBorrowedResponse);
  rpc borrow(BorrowRequest) returns (EmptyObject);
  rpc return_book(ReturnRequest) returns (EmptyObject);
}

message IsBorrowedRequest {
  int32 book_id = 1;
}

message IsBorrowedByRequest {
  int32 book_id = 1;
  int32 user_id = 2;
}

message IsBorrowedResponse {
  bool is_borrowed = 1;
} 

message BorrowRequest {
  int32 user_id = 1;
  int32 book_id = 2;
}

message ReturnRequest {
  int32 book_id = 1;
  int32 user_id = 2;
}

message EmptyObject {
}
